app/controllers/users_controller.rb

```diff
@@ -18,8 +21,33 @@ def new
     @user = User.new
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def index
+    @users = User.paginate page: params[:page]
+  end
+
   def show; end
 
+  def update
+    @user = User.find params[:id]
```

Comment:
Using find_by instead of find. Because `find` will raise exception if the record not found
```suggestion
    @user = User.find_by id: params[:id]
```

Reply:
done